### PR TITLE
feat: Initial version of wrapped MSSQL provider

### PR DESCRIPTION
**Pull request by tmeckel**
This PR contains the initial version of the Pulumi MSSQL provider, based in the [PGSSoft MSSQL Terraform provider](https://github.com/PGSSoft/terraform-provider-mssql)

### CODE REVIEW

**Comment by cobraz** [comment]
@tmeckel let me know if you want me to take a look :)

**Comment by tmeckel** [comment]
@cobraz The provider is still in a pre-alpha stage becauee of the new support for the Terraform Provider Framework. But if you like, go ahead a do a review on the current state. Input is always appreciated. 🤗

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+  # NPM REGISTRY THEN ENSURE THE NPM_REGISTRY_URL IS CHANGED
+  NUGET_PUBLISH_KEY: ${{ secrets.NUGET_PUBLISH_KEY }}
+  NUGET_FEED_URL: https://api.nuget.org/v3/index.json
+  PUBLISH_NUGET: true
+  # IF YOU NEED TO PUBLISH A PYPI PACKAGE THEN ENSURE AN PYPI_API_TOKEN
+  # SECRET IS SET AND PUBLISH_PYPI: TRUE. IF YOU WANT TO PUBLISH TO AN ALTERNATIVE
+  # PYPI REGISTRY THEN ENSURE THE PYPI_REPOSITORY_URL IS SET. IF YOU ARE USING AN API_TOKEN THEN
+  # YOU DO NOT NEED TO CHANGE THE PYPI_USERNAME (__token__) , IF YOU ARE USING PASSWORD AUTHENTICATION THEN YOU WILL
+  # NEED TO CHANGE TO USE THE CORRECT PASSWORD
+  PYPI_PASSWORD: ${{ secrets.PYPI_API_TOKEN }}
```
Comment (ringods) [inline .github/workflows/release.yml:32]: Should be `${{ secrets.PYPI_PASSWORD }}` for the org level secret to be picked up.

File: docs/installation-configuration.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+---
+title: MSSQL Setup
```
Comment (ringods) [inline docs/installation-configuration.md:2]: All over the Pulumi Registry, the title is standardized on `<provider> Installation & Configuration`. Please update.

File: provider/resources.go [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+	"github.com/pulumiverse/pulumi-mssql/provider/pkg/version"
+)
+
+//go:embed cmd/pulumi-resource-mssql/bridge-metadata.json
+var bridgeMetadata []byte
+
+// all of the token components used below.
+const (
+	// modules:
+	mainMod = "index" // the mssql module
```
Comment (ringods) [inline provider/resources.go:37]: Not a comment, rather a question: do you want all your resources and `get`-functions to end up in the top level module of every generated SDK? By leveraging other module names, you can group certain resources in submodules.

Example in `resources.go`:
https://github.com/pulumiverse/pulumi-unifi/blob/8240d3d4992fc76fc1676b71132c86f2be06c6ed/provider/resources.go#L38-L42

How this reflect in the SDKs & API docs:
https://www.pulumi.com/registry/packages/unifi/api-docs/

File: provider/resources.go [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+			PackageName: "pulumiverse_mssql",
+
+			// List any Python dependencies and their version ranges
+			Requires: map[string]string{
+				"pulumi": ">=3.0.0,<4.0.0",
+			},
+		},
+		Golang: &tfbridge.GolangInfo{
+			ImportBasePath: filepath.Join(
+				"github.com/pulumi/pulumi-mssql/sdk/",
```
Comment (ringods) [inline provider/resources.go:276]: Github organization name should be `pulumiverse` in `ImportBasePath`.

**Review by ringods — CHANGES_REQUESTED** [review]

File: provider/resources.go (hunk 3 above)
Comment (tmeckel) [reply provider/resources.go:37]: A fair question. However, the provider is so small that I don't think it's worth creating different modules.
image

Comment (ringods) [reply provider/resources.go:37]: 👍🏻

**Review by ringods — APPROVED** [review]

**Pull request merged**